#2559. Add more constructor augmentation tests around formal parameters

sgrekhov opened the pull request.

eernstg reviewed (commented):
Looks good! I made one suggestion, about a possible clarification of a description.

sgrekhov commented:
Updated. 
- Added test for implicit `dynamic`.
- In `augmenting_constructors_A01_t12.dart` description  added that the type is not cpecified explicitly
- Default values removed from augmentations
PTAL.

eernstg commented on LanguageFeatures/Augmentation-libraries/augmenting_constructors_A01_t12.dart:
```diff
@@ -0,0 +1,77 @@
+///
+/// @description Checks that it is not an error if a constructor augmentation
+/// specifies a type of formal parameter which were not specified in the
+/// introductory constructor.
```
Thanks!

eernstg reviewed (approved):
LGTM
